tests/queries/0_stateless/02494_parser_string_binary_literal.sql

```diff
@@ -0,0 +1,30 @@
+select b'';
+select b'0' == '\0';
```

Comment:
Why it's not equal? `select unbin('0') == '\0' -- returns 1`. The result should be the same as in `unbin` (and `unhex` for 'x' literal) function